src/transform/src/analysis/equivalences.rs

```diff
@@ -73,115 +77,165 @@ impl Analysis for Equivalences {
                         }
                     }
                 }
-                equivalences
+                Some(equivalences)
             }
             MirRelationExpr::Get { id, .. } => {
-                let mut equivalences = EquivalenceClasses::default();
+                let mut equivalences = Some(EquivalenceClasses::default());
                 // Find local identifiers, but nothing for external identifiers.
                 if let Id::Local(id) = id {
                     if let Some(offset) = depends.bindings().get(id) {
                         // It is possible we have derived nothing for a recursive term
                         if let Some(result) = results.get(*offset) {
                             equivalences.clone_from(result);
+                        } else {
+                            // No top element was prepared.
+                            // This means we are executing pessimistically,
+                            // but perhaps we must because optimism is off.
                         }
                     }
                 }
                 equivalences
             }
-            MirRelationExpr::Let { .. } => results.last().unwrap().clone(),
-            MirRelationExpr::LetRec { .. } => results.last().unwrap().clone(),
+            MirRelationExpr::Let { .. } => results.get(index - 1).unwrap().clone(),
+            MirRelationExpr::LetRec { .. } => results.get(index - 1).unwrap().clone(),
```

Comment:
```suggestion
            MirRelationExpr::Let { .. } => results[index - 1].clone(),
            MirRelationExpr::LetRec { .. } => results[index - 1].clone(),
```

Reply:
huh. weird. I have no explanation.

Reply:
Oh, I see. When going from `.last().unwrap()` it's pretty easy to just swap `.last()` for `.get(index)`. Easy to change, though typing this out I now feel dirty using `[]`.